File: algorithms/0007.reverse_integer/reverse_integer.rs
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        if x / 10 >= i32::MAX || x / 10 <= i32::MIN {
            return 0;
        }
        let mut number = x.abs() as i64;
        let mut result = 0i64;
        
        while number > 0 {
            let n = number % 10;
            result = result * 10 + n;
            number /= 10;
        }
        if result > i32::MAX as i64 || result < i32::MIN as i64 { return 0;}
        result as i32 * if x < 0 { -1 } else { 1 }
    }
}

File: algorithms/0007.reverse_integer/reverse_integer.cpp
class Solution {
public:
    int reverse(int x) {
        int ans = 0;
        while (x != 0) {
            if (x < 0 && ans < (INT_MIN - x%10)/10) {
                return 0;
            }
            if (x > 0 && ans > (INT_MAX - x%10)/10) {
                return 0;
            }
            ans = ans* 10 + x % 10;
            x /= 10;
        }
        return ans;
    }
};